minecode/tests/test_gnu.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# purldb is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/purldb for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+
+import os
+
+from django.test import TestCase
+from mock import patch
+
+from minecode.utils_test import JsonBasedTesting
+from minecode.visitors import gnu
+from packagedb.models import Package
+
+
+class GnuPriorityQueueTests(JsonBasedTesting, TestCase):
+    test_data_dir = os.path.join(os.path.dirname(__file__), "testfiles")
+
+    def setUp(self):
+        super(GnuPriorityQueueTests, self).setUp()
+        glibc_data_loc = self.get_test_loc("gnu/glibc/index.html")
+
+        with open(glibc_data_loc) as f:
+            self.glibc_data_content = f.read()
+
+    @patch("requests.get")
+    def test_process_request(self, mock_get):
+        mock_get.side_effect = [
+            type(
```

Comment:
Can you explain whats going on here?

Reply:
@JonoYang, `gnu.process_request` is using `fetchcode`, which in turn fetches and parses the GNU directory-listed source. During this process, `fetchcode` invokes `content` and `raise_for_status` on the response object. To mock this behavior, we dynamically create these response objects using `type`, with `content` and `raise_for_status` attributes.

Reply:
I see now, thanks!